scarlet-protocol-stomp/src/main/java/com/tinder/scarlet/stomp/okhttp/support/StompMessageDecoder.kt

```diff
@@ -0,0 +1,194 @@
+/*
+ * © 2018 Match Group, LLC.
+ */
+package com.tinder.scarlet.stomp.okhttp.support
+
+import com.tinder.scarlet.stomp.okhttp.models.StompCommand
+import com.tinder.scarlet.stomp.okhttp.models.StompMessage
+import java.io.ByteArrayOutputStream
+import java.nio.ByteBuffer
+
+/**
+ * An decoder for STOMP frames.
+ */
+class StompMessageDecoder {
+
+    /**
+     * Decodes the given byte array into a StompMessage.
+     * @param array the array to decode
+     */
+    fun decode(array: ByteArray): StompMessage? {
+        val byteBuffer = ByteBuffer.wrap(array)
```

Comment:
Have you consider using okio instead of `ByteBuffer`?

Reply:
I've rewritten decoder using okio instead of ByteBuffer